### PR TITLE
fix: handle gracefully downloadurls and srcsets with optional languages overwrite for files

### PR DESCRIPTION
**Pull request by skoegl**
See #1265

### CODE REVIEW

File: src/viur/core/modules/file.py [hunk 1]
```diff
@@ -630,7 +631,8 @@ def create_src_set(
         file: t.Union["SkeletonInstance", dict, str],
         expires: t.Optional[datetime.timedelta | int] = datetime.timedelta(hours=1),
         width: t.Optional[int] = None,
-        height: t.Optional[int] = None
+        height: t.Optional[int] = None,
+        language: t.Optional[str] = None
```
Comment (phorward) [inline src/viur/core/modules/file.py:635]:
```suggestion
        language: t.Optional[str] = None,
```

File: src/viur/core/modules/file.py [hunk 2]
```diff
@@ -658,6 +661,15 @@ def create_src_set(
...
             return ""
 
+        if isinstance(file, LanguageWrapper):
+            current_lang = current.language.get()
+            if language in file and file[language]:
+                file = file[language]
+            elif current_lang and current_lang in file:
+                file = file[current_lang]
+            else:
+                file = file[conf.i18n.default_language]
```
Comment (phorward) [inline src/viur/core/modules/file.py:671]: This should satisfy already, and returns empty when no language is given, but it is a LanguageWrapper.

```suggestion
            language = language or current.language.get()
            if not language or not (file := file.get(language)):
                return ""
```

The additional check on `conf.i18n.default_language` is already handled by `current.language.get()`, as this ContextVar is initialized with `conf.i18n.default_language` on request processing.

File: src/viur/core/render/html/env/viur.py [hunk 3]
```diff
@@ -679,7 +680,8 @@ def downloadUrlFor(
     fileObj: dict,
     expires: t.Optional[int] = conf.render_html_download_url_expiration,
     derived: t.Optional[str] = None,
-    downloadFileName: t.Optional[str] = None
+    downloadFileName: t.Optional[str] = None,
+    language: t.Optional[str] = None
```
Comment (phorward) [inline src/viur/core/render/html/env/viur.py:684]:
```suggestion
    language: t.Optional[str] = None,
```

File: src/viur/core/render/html/env/viur.py [hunk 4]
```diff
@@ -694,8 +696,19 @@ def downloadUrlFor(
...
     """
+
+    if isinstance(fileObj, LanguageWrapper):
+        current_lang = current.language.get()
+        if language in fileObj and fileObj[language]:
+            fileObj = fileObj[language]
+        elif current_lang and current_lang in fileObj:
+            fileObj = fileObj[current_lang]
+        else:
+            fileObj = fileObj[conf.i18n.default_language]
```
Comment (phorward) [inline src/viur/core/render/html/env/viur.py:710]: Same as above.

File: src/viur/core/render/html/env/viur.py [hunk 5]
```diff
@@ -733,7 +746,8 @@ def srcSetFor(
     fileObj: dict,
     expires: t.Optional[int] = conf.render_html_download_url_expiration,
     width: t.Optional[int] = None,
-    height: t.Optional[int] = None
+    height: t.Optional[int] = None,
+    language: t.Optional[str] = None
```
Comment (phorward) [inline src/viur/core/render/html/env/viur.py:750]:
```suggestion
    language: t.Optional[str] = None,
```

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @skoegl, thanks for creating this pull request.
Can you please test me suggested changes and apply it to the case below as well?

**Comment by skoegl** [comment]
Thank you for the code review . I hopefully have completed all requested changes. Please take another look and if ok, please merge.

**Review by phorward — APPROVED** [review]

**Pull request merged**